pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -935,6 +935,18 @@ func (s *DiscoveryServer) removeCon(conID string, con *XdsConnection) {
 	}
 }
 
+func (s *DiscoveryServer) workloadLabels(proxy *model.Proxy) model.Labels {
+	var workloadLabels model.Labels
+	ipAddresses := proxy.IPAddresses
+	if len(ipAddresses) == 0 {
```

Comment:
My thoughts were more if we have multiple ipAddress rather than none - I haven't had a chance to dig into how the addresses get set though so I am not sure what happens outside of k8s. @rshriram any ideas? I don't think its a major issue but I am only familiar with kubernetes

Reply:
I see.
Currently afaict only the kube registry feeds workloadLabels with the podIP.
However there could probably be more than 1 address in Proxy.IPAdresses and assuming that this PodIP is always in the first slot might be brittle.
Maybe it would make sense then to make workloadLabels return a LabelsCollection and require the workloadSelector to be a superset of every item in that collection.

What do you think?

Reply:
I think looking at all of them makes sense, the question I would have is do we need the workloadSelector to match *all* or *any* of the labels we get. I suspect we would want to do all. I think this is basically how it worked before right?

Reply:
Yes, that's how it worked before, just using the services labels.
I think it also makes sense to match against all Labels, with the current code I would assume that even with different IPAddresses only one set should be non-empty.
Thanks for the discussion! :+1:

Reply:
I just upgraded the PR to get the labels associated with all IP addresses.
The selector will match if it is a subset of any set of labels associated with a workloadId.
This is basically the same behavior as before, just that it tests against the labels of the Pods instead of the Services.

Reply:
cc @zhaohuabing any thoughts since you added the multi-ip support for each proxy, for consul environments

Reply:
I started working on an alternative fix today that would add a method `ServiceDiscovery.GetProxyWorkloadLabels()` which returns the workload labels for a proxy.
Maybe this works better for proxies not running in Kubernetes.
The change set will certainly be much, much larger.